### PR TITLE
Add options to ColorBarWidget to change shown checkboxes and their label/tooltip text

### LINKED ISSUE #16154: Add options to the ColorBarWidget, labels and tooltips should not always be SliceViewer-centric
The `ColorBarWidget` is a `MantidWidget` and is convenient in interfaces that are not necessarily used in the same context as the `SliceViewer`. The widget includes two "autoscale" tick boxes with labels, behavior and tooltips that are too specific to the `SliceViewer`. The widget is now used in the imaging interface which will need a bit of polishing. See #16131 for example.

- An option to change the tooltip texts, and maybe the label texts, will probably be good.
- An option to hide some or both tick boxes, simplifying the widget, could be considered.
- An option to modify the behavior of the tick boxes could be remotely considered but I think it won't be needed for imaging.

### PR DESCRIPTION
**Pull request by dtasev**
## Description of work.

The ColorBarWidget.cpp now has methods for changing both of the [checkboxes' label, tooltip text](https://github.com/mantidproject/mantid/commit/713a181a3bc51eb0b79dac859b0849e15402c8a3#diff-4aa05715847f2ffc407c1b4bd6c6ca59R381) and [visibility](https://github.com/mantidproject/mantid/commit/480f71ba1d4c22dbbb2931fef1201ea2e96ca792#diff-4aa05715847f2ffc407c1b4bd6c6ca59R108).
## To test:
- Go to `Interfaces` > `Diffraction` > `Tomographic Reconstruction`, go to tab `ROI etc.`
- Check that only **1** checkbox is shown, should be `Autoscale on load`, and that the tooltip is correct `"This flag signals that the color scale range will be set automatically to the current slice range when an image is loaded. Note that auto scaling will be applied when an image is loaded for the very first time."`
- Look for any spelling mistakes in the tooltip, I might've missed them
- Check that the Color Bar Widget behaves normally in the other places it's used (such as the Slice Viewer on a Workspace), and any other usages that I am not aware of.

Fixes #16154 .

_Does not need to be in the release notes._

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)? Is it well structured with small focussed classes/methods/functions?
- [ ] Are there unit/system tests in place? Are the unit tests small and test the a class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?
##### Functional Tests
- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] How do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant documentation been added/updated?
- [ ] Is user-facing documentation written in a user-friendly manner?
- [ ] Has developer documentation been updated if required?
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: MantidQt/MantidWidgets/inc/MantidQtMantidWidgets/ColorBarWidget.h [hunk 1]
```diff
@@ -56,11 +60,15 @@ class EXPORT_OPT_MANTIDQT_MANTIDWIDGETS ColorBarWidget : public QWidget {
   void setMaximum(double max);
   void setRenderMode(bool rendering);
 
+  /// Set which checkboxes are shown in the window
+  void setCheckBoxMode(CheckboxStrategy strategy);
```
Comment (FedeMPouzols) [inline MantidQt/MantidWidgets/inc/MantidQtMantidWidgets/ColorBarWidget.h]: Nice. don't hesitate to add bits of documentation like this when you see they're missing.

File: MantidQt/MantidWidgets/src/ColorBarWidget.cpp [hunk 2]
```diff
@@ -1,13 +1,10 @@
 #include "MantidQtAPI/MantidColorMap.h"
 #include "MantidQtMantidWidgets/ColorBarWidget.h"
 #include "MantidQtAPI/QScienceSpinBox.h"
-#include "qwt_scale_div.h"
 #include "MantidQtAPI/PowerScaleEngine.h"
-#include <iosfwd>
 #include <qwt_scale_map.h>
 #include <qwt_scale_widget.h>
 #include <QKeyEvent>
-#include <qwt_plot.h>
```
Comment (FedeMPouzols) [inline MantidQt/MantidWidgets/src/ColorBarWidget.cpp]: Good catch with these unused headers. Did you do it manually, or does ReSharper help?

Comment (DimitarTasev) [inline MantidQt/MantidWidgets/src/ColorBarWidget.cpp]: ReSharper with VS2015 grays out all unused headers, with a pretty good accuracy, at least for now.

**Comment by FedeMPouzols** [comment]
This looks good. It adds:
- Methods to modify the label text and the tooltip text of the two checkboxes that the `ColorBarWidget` has at the bottom (auto-scale and auto-scale for current slice).
- A method to change the availabitily/visibility of the auto-scale checkboxes.

The default behavior is the classic behavior of the SliceViewer, but these modifications are convenient to customize the appearance and behavior of the color bar widget for different interfaces. The color bar widget in the SliceViewer looks and works as usual. 

sv_screenshot


In the tomographic reconstruction / ROI tab the color bar widget is now more simple and the text(s) make better sense. The  option "autoscale on load" works well:

tomo_screen1

tomo_screen2


:shipit:

File: MantidQt/MantidWidgets/src/ColorBarWidget.cpp (hunk 2 above)
Comment (FedeMPouzols) [inline MantidQt/MantidWidgets/src/ColorBarWidget.cpp]: Interesting to know. This is probably interesting to other developers using MSVC.

**Pull request merged**